R/test_assumption.R

```diff
@@ -30,76 +44,32 @@ test_levene <- function(afex_aov, center = mean, ...){
 #' @param afex_aov \code{afex_aov} object.
 #'
 #' @author Mattan S. Ben-Shachar
+#' 
+#' @examples 
+#' \dontrun{
+#' fit_value <- aov_car(value ~ treatment * gender + Error(id/(phase*hour)), data = obk.long)
+#' 
+#' test_sphericity(fit_value)
+#' #>                             Test statistic p-value
+#' #> phase                              0.74927 0.27282
+#' #> treatment:phase                    0.74927 0.27282
+#' #> gender:phase                       0.74927 0.27282
+#' #> treatment:gender:phase             0.74927 0.27282
+#' #> hour                               0.06607 0.00760
+#' #> treatment:hour                     0.06607 0.00760
+#' #> gender:hour                        0.06607 0.00760
+#' #> treatment:gender:hour              0.06607 0.00760
+#' #> phase:hour                         0.00478 0.44939
+#' #> treatment:phase:hour               0.00478 0.44939
+#' #> gender:phase:hour                  0.00478 0.44939
+#' #> treatment:gender:phase:hour        0.00478 0.44939
+#' 
+#' }
 #'
 #' @export
 test_sphericity<- function(afex_aov){
-  summary(afex_aov)$sphericity.tests
-}
-
-#' Plot Residuals' QQ-plot
-#'
-#' Plot qq-plot and qq-line for the residuals of each error term. Used to test normality of residuals.
-#'
-#' @param afex_aov \code{afex_aov} object.
-#' @return if \code{return='plot'} will return a ggplot2 of the qqplot. Anything else will return a list of residulas.
-#'
-#' @author Mattan S. Ben-Shachar
-#'
-#' @export
-residuals_qqplot <- function(afex_aov, return = 'plot') {
-  if (!requireNamespace("ggplot2", quietly = TRUE)) {
-    stop("package ggplot2 is required.", call. = FALSE)
+  if (length(attr(between_1,'within'))==0) {
```

Comment:
It should probably not be `between_1` here. That suggests you did not check whether the function actually works before making the commit. 

To ensure functions work, all functionality should be accompanied by corresponding unit tests which can be found in the `tests/testthat` folder. I am preparing them for this functionality.